### PR TITLE
drivers: rtc: add support for max31343 real-time-clock

### PR DESCRIPTION
**Pull request by rbudai98**
Add support for https://www.analog.com/en/products/max31343.html

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [ ] I have build all projects affected by the changes in this PR
- [ ] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

File: drivers/rtc/max31343/max31343.c [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+/******************************************************************************/
+#include <stdlib.h>
+#include <errno.h>
+#include "max31343.h"
+#include "no_os_delay.h"
+#include "no_os_alloc.h"
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
```
Comment (CiprianRegus) [inline drivers/rtc/max31343/max31343.c:51]: Please remove these comments.

File: drivers/rtc/max31343/max31343.c [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+/**
+ * @brief Read device register.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param reg_data - The data read from the register.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int max31343_read(struct max31343_dev *dev, uint8_t reg_addr, uint8_t *reg_data)
```
Comment (CiprianRegus) [inline drivers/rtc/max31343/max31343.c:60]: I think this function should be named `max31343_reg_read`. That's what we usually do for the other drivers.

File: drivers/rtc/max31343/max31343.c [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+/**
+ * @brief Write device register.
+ * @param dev- The device structure.
+ * @param reg_addr - The register address.
+ * @param reg_data - The data to be written.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int max31343_write(struct max31343_dev *dev, uint8_t reg_addr, uint8_t reg_data)
+{
+	uint8_t buff[2] = { 0 };
```
Comment (CiprianRegus) [inline drivers/rtc/max31343/max31343.c:80]: You don't have to initialize this array

File: drivers/rtc/max31343/max31343.c [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+
+
+	/* I2C Initialization*/
+	ret = no_os_i2c_init(&dev->i2c_desc, init_param.i2c_init);
+	if (ret)
+		goto error_dev;
+
+	dev->battery_en = init_param.battery_en;
+
+	// Reset GPIO pin deassert
```
Comment (CiprianRegus) [inline drivers/rtc/max31343/max31343.c:138]: Please keep the same `/**/` comment style.

File: drivers/rtc/max31343/max31343.c [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+{
+	struct max31343_dev *dev;
+	int ret;
+
+	dev = (struct max31343_dev *)no_os_calloc(1, sizeof(*dev));
+	if (!dev)
+		return -ENOMEM;
+	*device = dev;
+
+
```
Comment (CiprianRegus) [inline drivers/rtc/max31343/max31343.c:130]: Extra empty line

File: drivers/rtc/max31343/max31343.c (hunk 1 above)
Comment (rbudai98) [reply drivers/rtc/max31343/max31343.c:51]: Why? As far as I see it is usually what in other drivers appears

**Comment by rbudai98** [comment]
v2:
* PR comm corrections

File: drivers/rtc/max31343/max31343.h [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+/* Trickle register bits */
+#define MAX31343_F_TRICKLE_D_TRICKLE		NO_OS_BIT(0)
+#define MAX31343_F_TRICKLE_TCHE			NO_OS_BIT(4)
+
+/* Temperature sensor config register bits */
+#define MAX31343_F_TS_CONFIG_TTSINT		NO_OS_BIT(3)
+#define MAX31343_F_TS_CONFIG_ONESHOT_MODE	NO_OS_BIT(6)
+#define MAX31343_F_TS_CONFIG_AUTO_MODE		NO_OS_BIT(7)
+
+#define MAX_31343_BCD2BIN(val)			(((val) & 15) + ((val) >> 4) * 10)
```
Comment (amiclaus) [inline drivers/rtc/max31343/max31343.h:131]: these macros should be generic. we don't want to have them duplicated in each rtc driver.
See:
https://github.com/torvalds/linux/blob/master/include/linux/bcd.h

Comment (rbudai98) [reply drivers/rtc/max31343/max31343.h:131]: Is it ok if I move these 2 defines into: no_os_util.h

**Comment by rbudai98** [comment]
v3:
* replaced bcd2bin and bin2bcd with defines from no_os_util.h

**Review by amiclaus — APPROVED** [review]

**Review by CiprianRegus — APPROVED** [review]

**Pull request merged**